app/bundles/LeadBundle/Tests/Controller/Api/ListApiControllerFunctionalTest.php

```diff
@@ -150,4 +150,49 @@ public function testBatchSegmentWorkflow()
             $this->assertIsArray($segment['filters']);
         }
     }
+
+    public function testSegmentWithCategory()
+    {
+        $categoryPayload = [
+            'title'  => 'API Cat',
+            'alias'  => 'kitty',
+            'bundle' => 'segment',
+        ];
+        $this->client->request('POST', '/api/categories/new', $categoryPayload);
+        $clientResponse     = $this->client->getResponse();
+        $response           = json_decode($clientResponse->getContent(), true);
+        $categoryId         = $response['category']['id'];
+
+        $segmentPayload = [
+            'name'        => 'API segment',
+            'description' => 'Segment created via API test',
+            'category'    => $categoryId,
+        ];
+
+        // Create:
+        $this->client->request('POST', '/api/segments/new', $segmentPayload);
+        $clientResponse = $this->client->getResponse();
+        $response       = json_decode($clientResponse->getContent(), true);
+        if (!empty($response['errors'][0])) {
+            $this->fail($response['errors'][0]['code'].': '.$response['errors'][0]['message']);
+        }
+
+        $segmentId = $response['list']['id'];
+
+        // Get segment with category by id:
+        $this->client->request('GET', "/api/segments/{$segmentId}");
+        $clientResponse = $this->client->getResponse();
+        $response       = json_decode($clientResponse->getContent(), true);
+
+        $this->assertSame(200, $clientResponse->getStatusCode());
+        $this->assertEquals($segmentPayload['category'], $response['list']['category']['id']);
+
+        // Search segments by category:
+        $this->client->request('GET', '/api/segments?search=category:kitty');
+        $clientResponse = $this->client->getResponse();
+        $response       = json_decode($clientResponse->getContent(), true);
+
+        $this->assertSame(200, $clientResponse->getStatusCode());
```

Comment:
There is a shorter way how to write the same:
```suggestion
        $this->assertTrue($clientResponse->isOk());
```